frontend/src/components/features/home/shared/generic-dropdown-menu.tsx

```diff
@@ -54,7 +54,16 @@ export function GenericDropdownMenu<T>({
 }: GenericDropdownMenuProps<T>) {
   const hasItems = filteredItems.length > 0;
   const showEmptyState = !hasItems && !stickyTopItem && !stickyFooterItem;
-
+  const getUniqueItemKey = (item: T): string => {
+    if (typeof item === "string") return item;
+    if (typeof item === "object" && item !== null && "id" in item) {
+      return String(item.id);
+    }
+    if (typeof item === "object" && item !== null) {
+      return String(Object.values(item)[0]);
```

Comment:
Issue:
Deriving a key from Object.values(item)[0] relies on property order and implicitly assumes the first value is unique and meaningful.

Impact:
Different objects may generate the same key, leading to collisions and unpredictable behavior.

Suggestion:
Do not infer uniqueness from arbitrary object properties. If an object does not have a well-defined identifier (e.g., id), the API should require one instead of guessing.